python/tvm/relax/op/transform.py

```diff
@@ -338,33 +343,130 @@ def full(
     dtype : Optional[str]
         The data type of the target.
 
+        If dtype is not given, the dtype of the target would be the dtype of fill_value.
+
     Returns
     -------
     result : relax.Expr
-        The resulting tensor.
+        The result tensor.
     """
-    if isinstance(shape, (PrimExpr, int)):
-        shape = [shape]
-    if isinstance(shape, (tuple, list)):
-        temp_shape = []
-        for shape in shape:
-            if isinstance(shape, PrimExpr):
-                temp_shape.append(shape)
-            elif isinstance(shape, int):
-                temp_shape.append(tvm.tir.const(shape, "int64"))
-            else:
-                raise RuntimeError(
-                    f"The input new shape of reshape operator contains unrecognized dimension {shape}"
-                )
-        shape = relax.ShapeExpr(temp_shape)
+    shape = _handle_shape(shape)
 
     if dtype is None:
-        dtype = tvm.DataType("void")
+        # currently tvm.DataType cannot identify "void"
+        # so leave dtype to be None so it will be converted into void type in PackedFunc calling process
+        # dtype = tvm.DataType("void")
+        dtype = None
```

Comment:
Yeah it probably was my mistake. I now know `None` will be converted into void but didn’t know this when writing that code. I think we can safely remove the this if-branch.